spec/models/activity_spec.rb

```diff
@@ -140,16 +140,20 @@
       let(:product) { activity.price_list.products.sample }
       let(:product_price) { activity.price_list.product_price_for(product).price }
       let(:cash_order) { FactoryBot.create(:order, :cash, activity: activity) }
+      let(:pin_order) { FactoryBot.create(:order, :pin, activity: activity) }
       let(:order) { FactoryBot.create(:order, activity: activity) }
 
       before do
         FactoryBot.create(:order_row, product: product, order: cash_order, product_count: 2)
+        FactoryBot.create(:order_row, product: product, order: pin_order, product_count: 2)
         FactoryBot.create(:order_row, product: product, order: order, product_count: 3)
       end
 
       it { expect(activity.revenue_with_cash).to eq product_price * 2 }
-      it { expect(activity.revenue_without_cash).to eq product_price * 3 }
-      it { expect(activity.revenue_with_cash).not_to eq activity.revenue_without_cash }
+      it { expect(activity.revenue_with_pin).to eq product_price * 2 * 0.981 }
```

Comment:
Please also check that the total revenue is 9* product_price (after changing pin oders to 4)

Reply:
You mean 4* right?